build_modules/lib/src/module_cache.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+
+import 'package:async/async.dart';
+import 'package:build/build.dart';
+
+import 'meta_module.dart';
+import 'modules.dart';
+
+final metaModuleCache = Resource<DecodingCache<MetaModule>>(
+    () => DecodingCache((m) => MetaModule.fromJson(m)),
+    dispose: (c) => c.dispose());
+
+final moduleCache = Resource<DecodingCache<Module>>(
+    () => DecodingCache((m) => Module.fromJson(m)),
+    dispose: (c) => c.dispose());
+
+class DecodingCache<T> {
```

Comment:
nit: add some doc comments describing this class

Reply:
Done.